nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1224 @@
+            return False
+        if xor(self.affine is None, other.affine is None):
+            return False
+        return (((self.affine is None and other.affine is None) or
+                 (abs(self.affine - other.affine).max() < 1e-8 and
+                  self.volume_shape == other.volume_shape)) and
+                (self.nvertices == other.nvertices) and
+                (self.arr == other.arr).all())
+
+    def __add__(self, other):
+        """
+        Concatenates two BrainModels
+
+        Parameters
+        ----------
+        other : BrainModel
+            brain model to be appended to the current one
+
+        Returns
+        -------
+        BrainModel
+        """
+        if type(self) == type(other):
+            if self.affine is None:
+                affine, shape = other.affine, other.volume_shape
+            else:
+                affine, shape = self.affine, self.volume_shape
+                if other.affine is not None and ((other.affine != affine).all() or
+                                                  other.volume_shape != shape):
+                    raise ValueError("Trying to concatenate two BrainModels defined in a different brain volume")
+            nvertices = dict(self.nvertices)
+            for name, value in other.nvertices.items():
+                if name in nvertices.keys() and nvertices[name] != value:
+                    raise ValueError("Trying to concatenate two BrainModels with inconsistent number of vertices for %s"
+                                     % name)
+                nvertices[name] = value
+            return type(self)(np.append(self.arr, other.arr), affine, shape, nvertices)
+        return NotImplemented
+
+
+class Parcels(Axis):
+    """
+    Each row/column in the CIFTI vector/matrix represents a parcel of voxels/vertices
+
+    This Axis describes which parcel is represented by each row/column.
+
+    Attributes
+    ----------
+    name : np.ndarray
+        (N, ) string array with the parcel names
+    parcel :  np.ndarray
+        (N, ) array with the actual get_parcels (each of which is a BrainModel object)
+
+    Individual get_parcels can also be accessed based on their name, using
+    ``parcel = parcel_axis[name]``
+    """
+    _use_dtype = np.dtype([('name', 'U60'), ('voxels', 'object'), ('vertices', 'object')])
+    _affine = None
+    _volume_shape = None
+
+    def __init__(self, arr, affine=None, volume_shape=None, nvertices=None):
+        """
+        Creates a new BrainModel axis defining the vertices and voxels represented by each row/column
+
+        Parameters
+        ----------
+        arr : np.ndarray
+            (N, ) structured array with for every element a tuple with 3 elements:
+            - string (name of parcel)
+            - (M, 3) int array with the M voxel indices in the parcel
+            - Dict[String -> (K, ) int array] mapping surface brain structure names to vertex indices
+        affine : np.ndarray
```

Comment:
same general comment as with the BrainModel class. I thin that @effigies should advise us on wether there's a preferred type of representation for axis elements than the typed array.